File: chatait-frontend-vue/src/utils/model/response/websocket.ts
/*
 * Copyright 2023 Anlity <leo@leocode.net>. All rights reserved.
 * Use of this source code is governed by a AGPL v3.0 style
 * license that can be found in the LICENSE file.
 */

import { ResponseConversationSpeakItemMjDataComponents } from '@/utils/model/response/conversation'

export interface WSMsg {
  type: string
  data: any
}

export interface WSConversationMidjourneyListenerEvent {
  conversation_id: string
  user_id: string
  topic_id: string
  topic_title: string
  topic_type: number
  role: string
  action_type: number
  content: string
  img_url: string
  thumbnail_img_url: string
  progress: number
  components: ResponseConversationSpeakItemMjDataComponents[]
  referenced_conversation_id: string
  referenced_components: ResponseConversationSpeakItemMjDataComponents[]
  error: string
}

File: chatait-frontend-server/app/model/response/websocket.go
// Copyright 2023 Anlity <leo@leocode.net>. All rights reserved.
// Use of this source code is governed by a AGPL v3.0 style
// license that can be found in the LICENSE file.

package response

import "github.com/anlityli/chatait-free/chatait-public-lib/library/api/midjourney"

type WebsocketConversationMidjourneyListenerEvent struct {
	ConversationId           string                                        `json:"conversation_id"`
	UserId                   string                                        `json:"user_id"`
	TopicId                  string                                        `json:"topic_id"`
	TopicTitle               string                                        `json:"topic_title"`
	TopicType                int                                           `json:"topic_type"`
	Role                     string                                        `json:"role"`
	ActionType               int                                           `json:"action_type"`
	Content                  string                                        `json:"content"`
	ImgUrl                   string                                        `json:"img_url"`
	ThumbnailImgUrl          string                                        `json:"thumbnail_img_url"`
	Progress                 int                                           `json:"progress"`
	Components               []*midjourney.WsReceiveMessageDComponentsItem `json:"components"`
	ReferencedConversationId string                                        `json:"referenced_conversation_id"`
	ReferencedComponents     []*midjourney.WsReceiveMessageDComponentsItem `json:"referenced_components"`
	Error                    string                                        `json:"error"`
}
